### ISSUE TITLE
Enforcing whitespace between at-rule and parentheses 

### ISSUE THREAD

**Issue by keegan-lillo** [kristerkari/stylelint-scss#140]
Similar to how this works: http://eslint.org/docs/rules/space-before-function-paren

There would be rules for: `@function`, `@mixin`, `@else if` (I think that's all that support parentheses)

### at-function-parentheses-space-before

Require or disallow a space before `@function` parentheses

```scss
@function foo ($arg) { }
/**                 ↑
 * The space before this parenthesis */
```

## Options

`string`: `"always"|"never"`

### `"always"`

There *must always* be exactly one space between the function name and the opening parenthesis

The following patterns are considered warnings:

```scss
@function foo($arg) { }
```
```scss
@function foo  ($arg) { }
```

The following patterns are *not* considered warnings:

```scss
@function foo ($arg) { }
```

### `"never"`

There *must never* be any whitespace between the function name and the opening parenthesis

The following patterns are considered warnings:

```scss
@function foo ($arg) { }
```

The following patterns are *not* considered warnings:

```scss
@function foo($arg) { }
```

**Comment by keegan-lillo** [comment]
edit: removed `@while` and `@if` because they are covered by `at-rule-name-space-after`

**Comment by kristerkari** [comment]
This was implemented in #141 :)
